src/NuGet.Clients/NuGet.VisualStudio.Implementation/NuGet.VisualStudio.Implementation.csproj

```diff
@@ -28,21 +28,22 @@
     <ProjectReference Include="$(NuGetClientsSrcDirectory)NuGet.VisualStudio.Common\NuGet.VisualStudio.Common.csproj" />
   </ItemGroup>
   <ItemGroup>
-    <PackageDownload Include="Microsoft.VisualStudio.ProjectSystem.Managed" />
-    <PackageDownload Include="Microsoft.VisualStudio.ProjectSystem.Managed.VS" />
+    <PackageReference Include="Microsoft.Build" IncludeAssets="None" PrivateAssets="all"/>
+    <PackageReference Include="Microsoft.Build.Framework" IncludeAssets="None" PrivateAssets="all"/>
+    <PackageReference Include="Microsoft.Build.Tasks.Core" IncludeAssets="None" PrivateAssets="all"/>
+    <PackageReference Include="Microsoft.Build.Utilities.Core" IncludeAssets="None" PrivateAssets="all"/>
+    <PackageReference Include="Microsoft.CodeAnalysis" IncludeAssets="None" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.CodeAnalysis.Common" IncludeAssets="None" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.CodeAnalysis.Features" IncludeAssets="None" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.CodeAnalysis.Workspaces.Common" IncludeAssets="None" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.VisualStudio.LanguageServices" IncludeAssets="None" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.VisualStudio.ProjectSystem.Managed" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.VisualStudio.ProjectSystem.Managed.VS" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.VisualStudio.ProjectSystem.VS" PrivateAssets="all" />
```

Comment:
I don't think this is an improvement over `PackageDownload` with assembly references. In fact, it's one line longer, despite the `<Reference` items being 3 lines each.

Reply:
I'm not a huge fan either. 
I think we should either reference a package version and let everything be done automatically by NuGet or keep using PackageDownload. 

Why do we have so many conflicts? Maybe we should talk/help the project system to fix those conflicts if they are on their end?

I know it's keeping things in multiple places, but I think the natural request is that PackageDownload is supported with CPM eventually in some way shape of form.

Reply:
What's the consensus here?  Should I have `<PackageDownload Update="" />` in `Directory.Build.targets` or should I use CPM like this?

Reply:
I think a way I can summarize this is that GeneratePathProperty and PackageDownload are both hacks. Ideally we just a normal PackageReference. Going from hack X to hack Y is not enough. I'm hoping we can remove the hacks.

> PackageDownload doesn't support GeneratePathProperty so you need to declare a property for the PackageReference and the Reference to keep things in sync. I personally don't find that to be better.

GeneratePathProperty for PackageDownload is on the backlog https://github.com/NuGet/Home/issues/8476. 
IMO, as long as you're doing something manual, you're failing. I don't find using 1 property instead of 2 is any better. 

We should aim to *not* use GeneratePathProperty. 

> PackageDownload doesn't resolve transitive dependencies so I'm guessing it was only used here as a way of getting around the package conflicts. That seems like a hack?

Yes, in general any usage of ExcludeAssets=all or GeneratePathProperty is also a hack, a sign that a package was either authored poorly or we have a weird shipping method.

Yes, I'd argue the same thing applies to specifying ExcludeAssets everywhere and using GeneratePathProperty.

> One of the references aren't even added as a PackageReference. Which means we are just relying on the fact that some other project is putting that package in our global packages folder. Using PackageReferences in a stock way helps avoid that.
 
 It relies on PackageDownload to get the package no?

> I certainly am not focused on number of lines of XML here. My focus was to use NuGet in a stock way and resolve package conflicts so that our product builds. The fact that I had to add all of this speaks to the customer pain that people go through all of the time. Isn't it a good thing for us to experience it in our own repo rather than work around it?

I think the customer pain here is because of an incorrect authored package. Similarly to what our customers would do, I think we should work with the package publishers to help us remove the hacks.

Reply:
I don't want to block this PR on that work ofc. Mostly just saying that this is a lateral move, rather than an improvement.

Reply:
I took a look and this is the only change leftover that I'm interested in. 

If you 2 prefer this approach, feel free to keep it. Happy to have an offline chat about it if needed.

Reply:
I'd prefer to keep the old `<PackageDownload`. We did it specifically because we want compile time APIs only. We don't run tests against their APIs, so we don't need their dependencies. We don't even need their API in our test's bin folder.

But from a pragmatic point of view, it doesn't matter so much. I think the difference is with `PackageDownload` when we have to increment the package version, it's one or two places we need to do it. With PackageReference, there's a whole lot extra work if project system's dependencies change. But that can always be the problem of the person who needs to bump the package version.

Reply:
We can leave it be then.

We should really talk to the project-system team and get them to publish their dependencies as well.